### PR TITLE
Adding youtube as a block

### PR DESCRIPTION
**Pull request by fullcolorcoder**
Adds the ability to embed/link to Youtube videos: 
Examples: 
- https://www.youtube.com/embed/LWJCbruwNC0
- https://www.youtu.be/LWJCbruwNC0

Resolves: [MWPW-115743](https://jira.corp.adobe.com/browse/MWPW-115743)

**Test URLs:**
- Before: https://main--milo--adobecom.hlx.page/drafts/bradjohn/youtube?martech=off
- After: https://youtube--milo--adobecom.hlx.page/drafts/bradjohn/youtube?martech=off

### CODE REVIEW

File: libs/blocks/youtube/youtube.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+      a.remove();
+    } else {
+      const id = embed.split('/').pop();
+      const embedHTML = `<div class="milo-video">
+      <iframe src="https://www.youtube.com${id ? `/embed/${id}?rel=0&amp;v=${id}` : embed}" webkitallowfullscreen mozallowfullscreen allowfullscreen style="border: 0; top: 0; left: 0; width: 100%; height: 100%; position: absolute;" allow="encrypted-media; accelerometer; gyroscope; picture-in-picture" scrolling="no" title="Content from Youtube" loading="lazy"></iframe>
+      </div>`;
+      a.insertAdjacentHTML('afterend', embedHTML);
+      a.remove();
+    }
+  }
```
Comment (chrischrischris) [inline libs/blocks/youtube/youtube.js:21]: nit: might be able to consolidate some of the common code into the first if block, something like:

```suggestion
if (a.origin?.includes('youtu')) {
  const id = embed.split('/').pop();
  let src;
  let className;
  if (embed.includes('embed')) {
    src = `https://www.youtube.com/embed/${id}`;
    className = 'youtube';
  } else {
    src = `https://www.youtube.com${id ? `/embed/${id}?rel=0&amp;v=${id}` : embed}`;
  }
  const embedHTML = `<div class="milo-video">
      <iframe src="${embedUrl}" ${className && `class="${className}"`} webkitallowfullscreen mozallowfullscreen allowfullscreen style="border: 0; top: 0; left: 0; width: 100%; height: 100%; position: absolute;" allow="encrypted-media; accelerometer; gyroscope; picture-in-picture" scrolling="no" title="Content from Youtube" loading="lazy">
      </iframe>
    </div>`;
  a.insertAdjacentHTML('afterend', embedHTML);
  a.remove();
}
```

**Review by chrischrischris — COMMENTED** [review]
Should we lazy load with intersection observer?

**Review by amauch-adobe — CHANGES_REQUESTED** [review]
Two concerns:

1. The page speed is really low on this PR, I have tried re-running the analysis multiple times and it keeps coming up as really low.  (27 for mobile, 76 for desktop).  This is no longer an issue @chrischrischris was gracious in correcting my mistaken thought and provided another example that proved we were good with the page speed.  https://youtube--milo--adobecom.hlx.page/drafts/cpeyer/youtube

2. My second concern is with the addition of the youtube block to the milo libs, I'm going to test this out against our SLA sites, but currently both blog.adobe.com and business.adobe.com/blog both have video embed blocks that allow youtube videos, and I worry about the confusion and or best practices that should be used.  This is more for teaching than anything.

**Comment by amauch-adobe** [comment]
@fullcolorcoder does the youtube block need to load in a modal or just as an embedded block?  I noticed the acceptance criteria said and/or the test page only shows loading it up as an embedded block.

Just wanted to make sure.

**Review by amauch-adobe — COMMENTED** [review]
@fullcolorcoder see my comments above, not necessarily changes that need to be made but just questions that will help me move this to approval.  

I have tested the above URLs on Firefox, Chrome, Safari, and Edge.  Also, mobile, and tablet responsiveness testing was successful.  Lastly, I verified the new block doesn't cause any issues with the current video blocks on our SLA sites, so we are good there as well.

File: libs/blocks/youtube/youtube.js (hunk 1 above)
Comment (fullcolorcoder) [reply libs/blocks/youtube/youtube.js]: Thanks, that greatly improved performance it looks like too.

**Comment by fullcolorcoder** [comment]
@amauch-adobe - yep, the AC in the ticket requires it to work in a modal. I should have included an example, here's one on a client example and our first use case for this. 

https://main--bacom--adobecom.hlx.page/customer-success-stories/coca-cola-case-study?milolibs=youtube#watch-now

**Review by amauch-adobe — APPROVED** [review]
QA approved, I have tested things out across browsers and platforms for both the modal test URL and the embedded test URL.  Everything looks good.

**Review by chrischrischris — APPROVED** [review]

**Pull request merged**